src/requests/typings/function.ts

```diff
@@ -158,9 +159,21 @@ export type AppEventTransformationResponse = {
 
 export type AppEventHandlerResponse = void
 
-export type AppActionRequest = {
+/**
+ * The app action request body will contain different properties depending on the schema of the app action
+ *
+ * Specify your app action schema as the generic type `Schema` to get the correct body type,
+ * e.g. `const { body: { message, recipient }} = event as AppActionRequest<'Notification.v1.0'>`
+ *
+ * If you are using a Custom schema, you can specify its shape as the second generic type `CustomSchemaBody`,
+ * e.g. `const { body: { myNumber }} = event as AppActionRequest<'Custom', { myNumber: number }>`
+ */
+export type AppActionRequest<
+  Schema extends AppActionSchema = 'Custom',
+  CustomSchemaBody = AppActionSchemaBodyMap['Custom'],
+> = {
   headers: Record<string, string | number>
-  body: Record<string, unknown>
+  body: Schema extends 'Custom' ? CustomSchemaBody : AppActionSchemaBodyMap[Schema]
```

Comment:
```suggestion
  body: AppActionSchemaBodyMap[Schema]
```
this doesn't work?

Reply:
I wanted to allow devs to optionally pass in their own types when using `Custom`, but it definitely increases the complexity, could back that out if it is over the top